pkg/kubelet/kubelet.go

```diff
@@ -503,7 +504,7 @@ func effectiveHairpinMode(hairpinMode componentconfig.HairpinMode, containerRunt
 			glog.Warningf("Hairpin mode set to %q but container runtime is %q, ignoring", hairpinMode, containerRuntime)
 			return componentconfig.HairpinNone, nil
```

Comment:
@bprashanth actually I'd like to wait until https://github.com/appc/cni/pull/175 lands and then just pass the request through to CNI.  Then we add a new plugin capability that kubenet advertises, that causes the docker Manager to skip hairpin.